vaadin-cdi/src/main/java/com/vaadin/cdi/context/RouteScopedContext.java

```diff
@@ -263,16 +263,16 @@ protected List<ContextualStorage> getActiveContextualStorages() {
 
     @Override
     protected ContextualStorage getContextualStorage(Contextual<?> contextual,
-            boolean createIfNotExist) {
-        RouteStorageKey key = convertToKey(contextual);
+                                                     boolean createIfNotExist) {
+        RouteStorageKey key = convertToKey(contextual, createIfNotExist);
         return contextManager.getContextualStorage(key, createIfNotExist);
     }
 
-    private RouteStorageKey convertToKey(Contextual<?> contextual) {
+    private RouteStorageKey convertToKey(Contextual<?> contextual, boolean createIfNotExist) {
         Bean<?> bean = getBean(contextual);
         UI ui = UI.getCurrent();
         Class<?> owner = getOwner(ui, bean);
-        if (!navigationChainHasOwner(ui, owner)) {
+        if (!navigationChainHasOwner(ui, owner) && createIfNotExist) {
             throw new IllegalStateException(String.format(
```

Comment:
So pertaining to my earlier comment: `return null` here and throw in `getContexstualStorage` after line 267 if `createIfNotExist=true` and `key=null`

Reply:
Thanks for suggestion.
Moving the check to the `getContexstualStorage` method means removing `convertToKey` at all, since to fill the exception message, we need all the code above.
Convert key will have only `contextManager.getKey(ui, owner)` and it doesn't make sense to have a method with this single line